File: CodeSpecification/filetemplete/Custom/KimView.xctemplate/UIViewObjective-C/___FILEBASENAME___.m
//___FILEHEADER___

#import "___FILEBASENAME___.h"

@interface ___FILEBASENAMEASIDENTIFIER___ ()

@end

@implementation ___FILEBASENAMEASIDENTIFIER___

#pragma mark - life cycle
- (instancetype)initWithFrame:(CGRect)frame {
    if (self = [super initWithFrame:frame]) {
        [self commonInit];
    }
    return self;
}

- (instancetype)initWithCoder:(NSCoder *)aDecoder {
    if (self = [super initWithCoder:aDecoder]) {
        [self commonInit];
    }
    return self;
}

- (id)initWithView:(UIView *)view {
    NSAssert(view, @"View must not be nil.");
    return [self initWithFrame:view.bounds];
}

#ifdef DEBUG
- (void)dealloc {
    NSLog(@"%s", __func__);
}
#endif

#pragma mark - Initial Methods
- (void)commonInit {
    /*custom view u want draw in here*/
    self.contentView.backgroundColor = [UIColor whiteColor];
    [self commonSetUpViews];
    [self commonLayoutViews];
}

- (void)commonSetUpViews {
    /*添加view*/
    [self.contentView addSubview:<#(nonnull UIView *)#>];
}

- (void)commonLayoutViews {
    /*约束view*/
    
}

#pragma mark - Layout
- (void)updateConstraints {
    /*更新约束，自定义view应该重写此方法在其中建立constraints.*/

    [super updateConstraints]; //this line must be the last line of this method.^_^
}

- (void)layoutSubviews {
    /*如果你需要更精确控制子view，而不是使用限制或autoresizing行为，就需要实现该方法。*/
}

#pragma mark - private method

#pragma mark - public Method
- (void)rendCellWith<#value#>:(<#value#>)<#value#> {
    <#value#>
}

#pragma mark - getters and setters

@end

File: CodeSpecification/filetemplete/Custom/KimTableViewCell.xctemplate/UITableViewCellSwift/___FILEBASENAME___.swift
//___FILEHEADER___

import UIKit

class ___FILEBASENAMEASIDENTIFIER___: ___VARIABLE_cocoaTouchSubclass___ {

    //MARK: - public property
    
    //MARK: - readonly property
    
    
    
    //MARK: - life cycle
    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)

        self.commonInit()
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        
        self.commonInit()
    }
    
    //MARK: - layout
    override func updateConstraints() {
        
        /*更新约束，自定义view应该重写此方法在其中建立constraints.*/
        
        super.updateConstraints() //this line must be the last line of this method.^_^
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        
        /*如果你需要更精确控制子view，而不是使用限制或autoresizing行为，就需要实现该方法。*/
    }

    #if DEBUG
    deinit {
        print("[\(type(of: self)) deinit]")
    }
    #endif
}


//MARK: - initial Methods
fileprivate extension ___FILEBASENAMEASIDENTIFIER___ {
    func commonInit() {
        self.contentView.backgroundColor = UIColor.white;
        self.commonSetUpViews()
        self.commonLayoutViews()
    }
    
    func commonSetUpViews() {
        <#function body#>
    }
    
    func commonLayoutViews() {
        <#function body#>
    }

}

//MARK: - public method
extension ___FILEBASENAMEASIDENTIFIER___ {
}

//MARK: - private method
fileprivate extension ___FILEBASENAMEASIDENTIFIER___ {
}

//MARK: - getters and setters
fileprivate extension ___FILEBASENAMEASIDENTIFIER___ {
    
}